File: mobile/lib/features/routes/data/route_comment.dart
class RouteComment {
  const RouteComment({
    required this.id,
    required this.username,
    required this.text,
    required this.createdAt,
  });

  final String id;
  final String username;
  final String text;
  final DateTime createdAt;

  factory RouteComment.fromJson(Map<String, dynamic> json) {
    return RouteComment(
      id: json['id'] as String? ?? '',
      username: json['username'] as String? ?? '',
      text: json['text'] as String? ?? '',
      createdAt: DateTime.tryParse(json['createdAt'] as String? ?? '') ??
          DateTime.fromMillisecondsSinceEpoch(0),
    );
  }
}

File: Rehberly.RouteService/Models/RouteComment.cs
namespace Rehberly.RouteService.Models;

public class RouteComment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid RouteId { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
